packages/gatsby/src/utils/page-data.ts

```diff
@@ -0,0 +1,76 @@
+import fs from "fs-extra"
+import path from "path"
+import { IGatsbyPage } from "../redux/types"
+import { store } from "../redux"
+
+interface IPageDataContext {
+  publicDir: string
+}
+
+type IPagePath = IGatsbyPage["path"]
+
+export const fixedPagePath = (pagePath: IPagePath): IPagePath =>
+  pagePath === `/` ? `index` : pagePath
+
+const getFilePath = (
+  { publicDir }: IPageDataContext,
+  pagePath: IPagePath
+): IPagePath =>
+  path.join(publicDir, `page-data`, fixedPagePath(pagePath), `page-data.json`)
+
+export const read = async (
+  { publicDir }: IPageDataContext,
+  pagePath: IPagePath
+): Promise<IGatsbyPage> => {
+  const filePath = getFilePath({ publicDir }, pagePath)
+  const rawPageData = await fs.readFile(filePath, `utf-8`)
+
+  return JSON.parse(rawPageData)
+}
+
+export const remove = async (
+  { publicDir }: IPageDataContext,
+  pagePath: IPagePath
+): Promise<void> => {
+  const filePath = getFilePath({ publicDir }, pagePath)
+
+  if (fs.existsSync(filePath)) {
+    return await fs.remove(filePath)
+  }
+
+  return Promise.resolve()
+}
+
+export const write = async (
+  { publicDir }: IPageDataContext,
+  { componentChunkName, matchPath, path }: IGatsbyPage,
+  result: any
+): Promise<void> => {
+  const filePath = getFilePath({ publicDir }, path)
+  const body = {
+    componentChunkName,
+    path,
+    matchPath,
+    result,
+  }
+  const bodyStr = JSON.stringify(body)
+  // transform asset size to kB (from bytes) to fit 64 bit to numbers
+  const pageDataSize = Buffer.byteLength(bodyStr) / 1000
+
+  store.dispatch({
+    type: `ADD_PAGE_DATA_STATS`,
+    payload: {
+      filePath,
+      size: pageDataSize,
+    },
+  })
+
+  await fs.outputFile(filePath, bodyStr)
+}
+
+export default {
+  fixedPagePath,
+  read,
+  remove,
+  write,
+}
```

Comment:
Added a redundancy in the file exports to ensure the compatibility of the import statements in existing JavaScript files using the default statement in a `require` statement.

Reply:
Let's remove this and just update all the references to this file!